src/hooks/useShortcut.js

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useCallback, useReducer } from 'react'
+
+const keysReducer = (state, action) => {
+  switch (action.type) {
+    case 'set-key-down':
+      const keydownState = { ...state, [action.key]: true }
+      return keydownState
+    case 'set-key-up':
+      const keyUpState = { ...state, [action.key]: false }
+      return keyUpState
+    case 'reset-keys':
+      const resetState = { ...action.data }
+      return resetState
+    default:
+      return state
+  }
+}
+
+const useShortcut = (shortcutKeys, callback) => {
+  if (!Array.isArray(shortcutKeys))
+    throw new Error(
+      'The first parameter to `useKeyboardShortcut` must be an ordered array of `KeyboardEvent.key` strings.'
+    )
+
+  if (!shortcutKeys.length)
+    throw new Error(
+      'The first parameter to `useKeyboardShortcut` must contain atleast one `KeyboardEvent.key` string.'
+    )
+
+  if (!callback || typeof callback !== 'function')
```

Comment:
It should be enough to check if it is not a function. Everything falsy is not a function too.

```suggestion
  if (typeof callback !== 'function')
```